extension/lib/settings_wrapper.js

```diff
@@ -103,127 +134,83 @@ class Settings {
 	// }
 
 	// TODO: Track storage left, when nearing capacity do something
+	// See DOCUMENT_FREQ
 	setDocumentFreq(documentFreq) {
-		chrome.storage.local.set({[settingKey.DOCUMENT_FREQ]: documentFreq}, function() {
+		let key = settingKey.DOCUMENT_FREQ;
+		this.settings[key] = documentFreq;
+		chrome.storage.local.set({[key]: documentFreq}, function() {
 			if (chrome.runtime.lastError) {
 				console.log("Failed to save term document frequencies: " + chrome.runtime.lastError);
 				return;
 			}
 		});
 	}
 
-	// Set the URLs that the user has visited
+	// See VISITED_URLS
 	setVisitedUrls(visitedUrls) {
-		chrome.storage.local.set({[settingKey.VISITED_URLS]: visitedUrls}, function() {
+		let key = settingKey.VISITED_URLS;
+		this.settings[key] = visitedUrls;
+		chrome.storage.local.set({[key]: visitedUrls}, function() {
 			if (chrome.runtime.lastError) {
 				console.log("Failed to save visited URLs list: " + chrome.runtime.lastError);
 				return;
 			}
 		});
 	}
 
-
-
-	/*
-	See setSpeed().
-	If not set yet, return default value of 20.
-	Parameters;
-	- cb: func(int). A callback function which accepts the retrieved speed.
-	*/
+	// See SPEED
 	getSpeed(cb) {
 		let key = settingKey.SPEED;
-		let val = chrome.storage.local.get(key, function(settingsDict) {
-			if (chrome.runtime.lastError) {
-				console.log("Failed to retrieve speed setting: " + chrome.runtime.lastError);
-				return;
-			}
+		if (!(key in this.settings)) {
 			// Default speed.
-			let speed = 260;
-			if (key in settingsDict) {
-				speed = settingsDict[key];
-			} 
-			cb(speed);
-		})
+			this.settings[key] = 260;
+		}
+		return this.settings[key];
 	}
-	// See setDefaultTrackerSettigns().
-	getDefaultTrackerSettings() {
-		return {
-			[trackerSettingKey.KEYWORD] : trackerSettingValue.GREEN,
-			[trackerSettingKey.HIGHLIGHTER]: trackerSettingValue.BLUE,
-			[trackerSettingKey.SHADOW]: trackerSettingValue.BLUE
-		};
-	}	
-	
-
-	/*
-	See setCustomizations().
-	If not set yet, return getDefaultTrackerSettings()
-	Parameters;
-	- cb: func({trackerSettingKey : trackerSettingValue}).
-	  A callback function which accepts the retrieved customizations status.
-	*/
-	getCustomizations(cb) {
+
+	// See TRACKER_CUSTOM
+	getCustomizations() {
 		let key = settingKey.TRACKER_CUSTOM;
-		let that = this;
-		let val = chrome.storage.local.get(key, function(settingsDict) {
-			if (chrome.runtime.lastError) {
-				console.log("Failed to retrieve customization setting: " + chrome.runtime.lastError);
-				return;
-			}
-			let customs = that.getDefaultTrackerSettings();
-			if (key in settingsDict) {
-				customs = settingsDict[key];
-			}
-			cb(customs);
-		});
+		if (!(key in this.settings)) {
+			// Default customization
+			this.settings[key] = {
+				[trackerSettingKey.KEYWORD] : trackerSettingValue.GREEN,
+				[trackerSettingKey.HIGHLIGHTER]: trackerSettingValue.BLUE,
+				[trackerSettingKey.SHADOW]: trackerSettingValue.BLUE
```

Comment:
When merge with master this will need to be changed to:

[trackerSettingKey.KEYWORD] : trackerSettingValue.LIGHT

(We don't use highlighter or shadow anymore; "Light" is just what I've been going with as the default, but can see what users prefer it to start w)